### PR TITLE
Internal issue roda-impl/roda-intrasoft#189

### PR DESCRIPTION
**Pull request by AntonioG70**

### CODE REVIEW

File: roda-core/roda-core/src/main/java/org/roda/core/common/monitor/TransferredResourcesScanner.java [hunk 1]
```diff
@@ -311,7 +313,7 @@ public String renameTransferredResource(TransferredResource resource, String new
         }
       }
 
-      Path relativeToBase = basePath.relativize(resourcePath.getParent().resolve(newName));
+      Path relativeToBase = basePath.relativize(resourcePath.getParent().resolve(FilenameUtils.normalize(newName)));
```
Comment (006627) [inline roda-core/roda-core/src/main/java/org/roda/core/common/monitor/TransferredResourcesScanner.java:316]: be careful with relative paths, test your changes

File: roda-core/roda-core/src/main/java/org/roda/core/common/validation/ResourceResolver.java [hunk 2]
```diff
@@ -40,10 +41,10 @@ public byte[] load(String href) throws Exception {
       try {
         String filename = href;
         try {
-          filename = Paths.get(URI.create(href)).getFileName().toString();
+          filename = FilenameUtils.normalize(Paths.get(URI.create(href)).getFileName().toString());
```
Comment (006627) [inline roda-core/roda-core/src/main/java/org/roda/core/common/validation/ResourceResolver.java:44]: This is a file name use FilenameUtils.getName

**Review by 006627 — CHANGES_REQUESTED** [review]

File: roda-core/roda-core/src/main/java/org/roda/core/RodaCoreFactory.java [hunk 3]
```diff
@@ -656,7 +657,7 @@ private static void initializeWorkingDirectory() {
       String systemTmpDir = getSystemProperty("java.io.tmpdir", "tmp");
       Path defaultRodaWorkingDirectory = Files.createTempDirectory(Paths.get(systemTmpDir), "rodaWorkingDirectory");
       workingDirectoryPath = Paths
-        .get(getRodaConfiguration().getString("core.workingdirectory", defaultRodaWorkingDirectory.toString()));
+        .get(getRodaConfiguration().getString("core.workingdirectory", FilenameUtils.normalize(defaultRodaWorkingDirectory.toString())));
       Files.createDirectories(workingDirectoryPath);
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply roda-core/roda-core/src/main/java/org/roda/core/RodaCoreFactory.java:661]: The method is working correctly.

File: roda-core/roda-core/src/main/java/org/roda/core/common/monitor/ReindexTransferredResourcesRunnable.java [hunk 4]
```diff
@@ -58,7 +59,7 @@ public void run() {
       EnumSet<FileVisitOption> opts = EnumSet.of(FileVisitOption.FOLLOW_LINKS);
       Path path;
       if (folderRelativePath.isPresent()) {
-        Path resolvedBasePath = basePath.resolve(Paths.get(folderRelativePath.get()));
+        Path resolvedBasePath = basePath.resolve(Paths.get(FilenameUtils.normalize(folderRelativePath.get())));
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply roda-core/roda-core/src/main/java/org/roda/core/common/monitor/ReindexTransferredResourcesRunnable.java:62]: The method is working correctly.

File: roda-core/roda-core/src/main/java/org/roda/core/common/monitor/TransferredResourcesScanner.java [hunk 5]
```diff
@@ -224,7 +226,7 @@ public Optional<String> updateTransferredResources(Optional<String> folderRelati
     Path resolvedBasePath;
     Optional<String> normalizedRelativePath;
     if(folderRelativePath.isPresent() && !folderRelativePath.get().isEmpty()) {
-      resolvedBasePath = basePath.resolve(Paths.get(folderRelativePath.get())).normalize();
+      resolvedBasePath = basePath.resolve(Paths.get(FilenameUtils.normalize(folderRelativePath.get()))).normalize();
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply roda-core/roda-core/src/main/java/org/roda/core/common/monitor/TransferredResourcesScanner.java:229]: The method is working correctly.

File: roda-core/roda-core/src/main/java/org/roda/core/common/validation/ResourceResolver.java (hunk 2 above)
Comment (AntonioG70) [reply roda-core/roda-core/src/main/java/org/roda/core/common/validation/ResourceResolver.java]: Done

File: roda-common/roda-common-utils/src/main/java/org/roda/core/util/ZipUtility.java [hunk 6]
```diff
@@ -125,15 +126,15 @@ private static List<File> extractFilesFromInputStream(InputStream inputStream, F
 
       while (zipEntry != null) {
         // for each entry to be extracted
-        String entryName = zipEntry.getName();
+        String entryName = FilenameUtils.normalize(zipEntry.getName());
```
[the remark this answers is not in the document]
Comment (AntonioG70) [reply roda-common/roda-common-utils/src/main/java/org/roda/core/util/ZipUtility.java]: Done

**Pull request merged**